src/Apps/Collect/Components/Collection/Header/DefaultHeader.tsx

```diff
@@ -0,0 +1,149 @@
+import { DefaultHeader_headerArtworks } from "__generated__/DefaultHeader_headerArtworks.graphql"
+import React, { FC } from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import styled from "styled-components"
+import { useWindowSize } from "Utils/Hooks/useWindowSize"
+import { getViewportWidth } from "Utils/viewport"
+
+interface Props {
+  headerArtworks: DefaultHeader_headerArtworks
+  defaultHeaderImageHeight: number
+}
+
+const IMAGE_MARGIN_X = 10
+const LARGE_VIEWPORT_WIDTH = 880
+
+export const CollectionDefaultHeader: FC<Props> = ({
+  headerArtworks,
+  defaultHeaderImageHeight,
+}) => {
+  useWindowSize()
+  const { hits: artworks } = headerArtworks
+
+  if (!artworks) {
+    return null
+  }
+
+  const viewportWidth = getViewportWidth()
+  const smallViewport = viewportWidth < LARGE_VIEWPORT_WIDTH
+  const duplicatedArtworks = artworks.slice(0)
+  const artworksToRender = getHeaderArtworks(
+    duplicatedArtworks,
+    viewportWidth,
+    smallViewport
+  )
+
+  return (
+    <header>
+      <DefaultHeaderContainer
+        position={["relative", "absolute"]}
+        left={["auto", 0]}
+        width={["auto", 1]}
+        height={[160, 160, 250]}
+      >
+        <HeaderArtworks>
+          {artworksToRender.map((artwork, i) => {
+            return (
+              <a href={artwork.href} key={i}>
+                <Image
+                  mx={0.5}
+                  height={defaultHeaderImageHeight}
+                  src={
+                    smallViewport
+                      ? artwork.image.small.url
+                      : artwork.image.large.url
+                  }
+                  preventRightClick
+                />
+              </a>
+            )
+          })}
+        </HeaderArtworks>
+      </DefaultHeaderContainer>
+    </header>
+  )
+}
+
+export const getHeaderArtworks = (
+  originalArtworksArray: any[],
+  headerWidth: number,
+  isSmallViewport: boolean
+) => {
+  let artworkWidthsInOriginalArray = 0
+  let artworkWidthsInRepeatingArray
+  const repeatedArtworksArray = []
+
+  originalArtworksArray.forEach(artwork => {
+    isSmallViewport
+      ? (artworkWidthsInOriginalArray +=
+          artwork.image.small.width + IMAGE_MARGIN_X)
+      : (artworkWidthsInOriginalArray +=
+          artwork.image.large.width + IMAGE_MARGIN_X)
+    repeatedArtworksArray.push(artwork)
+  })
+
+  artworkWidthsInRepeatingArray = artworkWidthsInOriginalArray
+
+  /**
+   * If the summed widths of the artworks returned is larger than the width
+   * of the viewport then there's no need to duplicate/repeat the header's artworks
+   */
+  if (artworkWidthsInOriginalArray >= headerWidth) {
+    return originalArtworksArray
+  } else {
+    /**
+     * Otherwise, duplicate the original array of artworks and add a new artwork to the array
+     * until the summed widths of the artworks is greater than the viewport width
+     */
+    while (artworkWidthsInRepeatingArray <= headerWidth) {
```

Comment:
Nicely done 👏🏾 My only feedback is there's a lot of code duplication between this while block and those lines above. May make sense to extract into a local helper function

https://github.com/artsy/reaction/pull/2634/files#diff-224a40bc680d52b38baeff65c547243eR77-R84